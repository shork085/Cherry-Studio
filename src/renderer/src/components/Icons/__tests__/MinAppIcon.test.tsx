import { render } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'

import MinAppIcon from '../MinAppIcon'

vi.mock('@renderer/config/minapps', () => ({
  DEFAULT_MIN_APPS: [
    {
      id: 'test-app-1',
      name: 'Test App 1',
      logo: '/test-logo-1.png',
      url: 'https://test1.com',
      bodered: true,
      background: '#f0f0f0'
    },
    {
      id: 'test-app-2',
      name: 'Test App 2',
      logo: '/test-logo-2.png',
      url: 'https://test2.com',
      bodered: false,
      background: undefined
    }
  ]
}))

describe('MinAppIcon', () => {
  const mockApp = {
    id: 'test-app-1',
    name: 'Test App',
    url: 'https://test.com',
    style: {
      opacity: 0.8,
      transform: 'scale(1.1)'
    }
  }

  it('should render correctly with various props', () => {
    const customStyle = { marginTop: '10px' }
    const { container } = render(<MinAppIcon app={mockApp} size={64} style={customStyle} sidebar={false} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should not apply app.style when sidebar is true', () => {
    const { container } = render(<MinAppIcon app={mockApp} sidebar={true} />)
    const img = container.querySelector('img')

    expect(img).not.toHaveStyle({
      opacity: '0.8',
      transform: 'scale(1.1)'
    })
  })

  it('should return null when app is not found in DEFAULT_MIN_APPS', () => {
    const unknownApp = {
      id: 'unknown-app',
      name: 'Unknown App',
      url: 'https://unknown.com'
    }
    const { container } = render(<MinAppIcon app={unknownApp} />)

    expect(container.firstChild).toBeNull()
  })
})
