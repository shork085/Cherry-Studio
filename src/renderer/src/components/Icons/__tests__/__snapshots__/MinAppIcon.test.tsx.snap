// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MinAppIcon > should render correctly with various props 1`] = `
.c0 {
  border-radius: 16px;
  user-select: none;
  -webkit-user-drag: none;
}

<img
  class="c0"
  src="/test-logo-1.png"
  style="border: 0.5px solid var(--color-border); width: 64px; height: 64px; background-color: rgb(240, 240, 240); opacity: 0.8; transform: scale(1.1); margin-top: 10px;"
/>
`;
